ViewControllers/SearchViewController.swift

```diff
@@ -0,0 +1,123 @@
+//
+
+import UIKit
+
+class SearchViewController: UIViewController {
+    let movieDataService = MovieDataService()
+    private var searchResults = [Movie]()
+    private var isSearching = false
+    
+    private lazy var searchController: UISearchController = {
+       let search = UISearchController(searchResultsController: nil)
+        search.searchResultsUpdater = self
+        search.obscuresBackgroundDuringPresentation = false
+        search.automaticallyShowsCancelButton = true
+        search.hidesNavigationBarDuringPresentation = false
+        search.searchBar.placeholder = "Type movie name"
+        search.searchBar.tintColor = UIColor(named: "titleColor")
+//        This option doesn't work from here but works OK from loadView()
+//        search.searchBar.searchTextField.textColor = UIColor(named: "titleColor")!
+        
+//        This option works but looks bulky and changes the size of the textField
+//        UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).defaultTextAttributes = [NSAttributedString.Key.foregroundColor: UIColor(named: "titleColor")]
+        return search
+    }()
+    
+    private lazy var tableView: UITableView = {
+       let tableView = UITableView()
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.backgroundColor = UIColor(named: "backgroundColor")
+        tableView.register(SearchTableViewCell.self)
+        tableView.rowHeight = 124
+        tableView.dataSource = self
+        tableView.delegate = self
+        return tableView
+    }()
+    
+    override func loadView() {
+        super.loadView()
+        view.backgroundColor = UIColor(named: "backgroundColor")
+        
+        title = "Search Movies"
+        navigationController?.navigationBar.barTintColor = UIColor(named: "backgroundColor")
+        navigationController?.navigationBar.titleTextAttributes = [.foregroundColor: UIColor.white, .font: UIFont.systemFont(ofSize: 20, weight: .heavy)]
+        
+        navigationItem.searchController = searchController
+        searchController.searchBar.searchTextField.textColor = UIColor(named: "titleColor")!
+        
+        view.addSubview(tableView)
+        NSLayoutConstraint.activate([
+            tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ])
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let notificationCenter = NotificationCenter.default
+        notificationCenter.addObserver(self, selector: #selector(makeRequest), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    @objc private func makeRequest() {
```

Comment:
Code encapsulation is not only about private/public, sometimes we can leak functionality through method names. In this case here, our method is called `makeRequest` which makes complete sense as we are requesting it through an API but what if we change that to a local database? In this case it won't make sense anymore.
Since this method is invoked when the user finishes editing, we could name it: `didFinishEditing` or `shouldStartSearching` for example.